Configurable time limit

davidoskky opened the pull request:
## 📑 Configurable time limit
Adds an option in the config file to set how many minutes should the game last.
15 minutes is not enough with lots of guns.

Moreover, do not count warmup time in game time and display how much warmup time is missing in the scoreboard before the start of the match.

## ✅ Checks
<!-- Make sure your pr passes the CI checks and do check the following fields as needed - -->
- [x] My pull request adheres to the code style of this project.
- [ ] My code requires changes to the documentation.
- [ ] I have updated the documentation, comments as required.
- [x] I've tested my commits, they work and do not send users to oblivion.

davidoskky committed:
Add the TimeLimit setting

davidoskky committed:
Display how much time is left before warmup ends

Shtoyan commented:
Woah, thanks for PR.

But your changes for scoreboard doesn't work for me on dedicated server. Did you test it only in solo?

davidoskky commented:
Indeed; I'm sorry.
I'll try later to understand what is not replicating on the server.
I guess it's the `bDoingWarmup`.

davidoskky committed:
Replicate warmup time

davidoskky commented:
I could not test it on my dedicated server yet, but this looks like it should work.

Shtoyan commented:
Sorry for the late reply. I just tested - Warmup time shows correctly, but not the remaining time after main game starts: aka it's TimeLimit - WarmupTime.

davidoskky commented:
Alright, don't worry. I have got it on my server for testing now and on I'll send you the fixed version either today or tomorrow.

Shtoyan commented:
@davidoskky hi, do you still work on this or by any chance had finished it?

davide-grheco committed:
Fix remaining time display after warmup and add build Makefile

The time limit countdown ran during warmup, so RemainingTime was already depleted by WarmupTime seconds when the main game started. Warmup now resets both GRI and GameInfo RemainingTime to TimeLimit * 60.

Also fix the scoreboard to read ElapsedTime and RemainingTime from GG_GRI so values are correctly replicated to clients on dedicated servers.

davide-grheco committed:
Use WarmupCountDown rather than custom calculation

davidoskky commented:
Sorry, I had completely forgot about this. Apparently I had already implemented the fix but never tested it.

I could only test it locally as currently for some reason my local KF can not connect to any server for some reason. Looking at the changes this should work on a dedicated server.
